packages/widgets/src/media-server/component.tsx

```diff
@@ -83,49 +93,147 @@ export default function MediaServerWidget({ integrationIds, isEditMode }: Widget
     mantineTableProps: {
+      className: "media-server-widget-table",
       style: {
         tableLayout: "fixed",
       },
     },
     mantineTableContainerProps: {
       style: {
-        flexGrow: 5,
+        height: "100%",
       },
     },
+    mantineTableBodyCellProps: ({ row }) => ({
+      onClick: () => {
+        setClickedIndex(row.index);
+        open();
+      },
+    }),
   });
 
   return (
-    <Box h="100%">
+    <Stack gap={0} h="100%" display="flex" style={baseStyle}>
       <MantineReactTable table={table} />
-    </Box>
+      {flatSessions[0]?.integrationKind && (
+        <Group
+          gap="1cqmin"
+          h="var(--ratio-width)"
+          px="var(--space-size)"
+          pr="5cqmin"
+          justify="flex-end"
+          style={{
+            borderTop: "0.0625rem solid var(--border-color)",
+          }}
+        >
+          <Image
+            className="media-server-icon"
+            src={integrationDefs[flatSessions[0].integrationKind].iconUrl}
+            w="sm"
+            h="sm"
+            fit="contain"
+          />
+          <Text className="media-server-name" size="sm">
+            {integrationDefs[flatSessions[0].integrationKind].name}
+          </Text>
+        </Group>
+      )}
+      <ItemInfoModal items={flatSessions} currentIndex={clickedIndex} opened={opened} onClose={close} />
+    </Stack>
   );
 }
+
+interface ItemInfoModalProps {
+  items: StreamSession[];
+  currentIndex: number;
+  opened: boolean;
+  onClose: () => void;
+}
+
+const ItemInfoModal = ({ items, currentIndex, opened, onClose }: ItemInfoModalProps) => {
+  const t = useScopedI18n("widget.mediaServer.items");
+  const item = useMemo<StreamSession | undefined>(() => items[currentIndex], [items, currentIndex]);
+  const currentlyPlayingType =
+    item?.currentlyPlaying?.type === "movie" ? (
+      <IconMovie size={20} />
+    ) : item?.currentlyPlaying?.type === "tv" ? (
+      <IconDeviceTv size={20} />
+    ) : item?.currentlyPlaying?.type === "video" ? (
+      <IconVideo size={20} />
+    ) : item?.currentlyPlaying?.type === "audio" ? (
+      <IconDeviceAudioTape size={20} />
+    ) : null;
+
+  return (
+    <Modal opened={opened} onClose={onClose} centered title={currentlyPlayingType} size="auto">
+      {item === undefined ? (
+        <Center>{"No item found"}</Center>
+      ) : (
+        <Stack align="center">
+          <Flex direction="column" gap="xs" align="center">
+            <Title>{item.currentlyPlaying?.name}</Title>
+            <Title order={3}>
+              {item.currentlyPlaying?.episodeName} (
+              {item.currentlyPlaying?.seasonName && item.currentlyPlaying.seasonName})
+            </Title>
+          </Flex>
+          <NormalizedLine itemKey={t("user")} value={item.user.username} />
+          <NormalizedLine itemKey={t("name")} value={item.sessionName} />
+          <NormalizedLine itemKey={t("id")} value={item.sessionId} />
+        </Stack>
+      )}
+    </Modal>
```

Comment:
Instead of using the general Modal component you could use the modal system we have. Then you can also directly pass in innerProps and don't need to use the index and `useDisclosure`.

See `copy-api-key-modal.tsx` for an example. You will need the `createModal` method AND a `withOptions` after calling that function. If something is not clear, just ask!

Reply:
I build it like downloads widget, and there is not use with createModal. Why not to use Mantine modal component?

Reply:
Under the hood it's also using that, because then you don't need to render it in the dom yourself and can just call a function after using the useModalAction hook

Reply:
need help with this